packages/scoped-custom-element-registry/test/Element.test.html.js

```diff
@@ -104,4 +109,41 @@ describe('Element', () => {
       expect($el.firstElementChild).to.not.be.instanceof(CustomElementClass);
     });
   });
+
+  describe('attributes', () => {
+    it('should call setAttribute', () => {
+      const {tagName, CustomElementClass} = getObservedAttributesTestElement(['foo']);
+      customElements.define(tagName, CustomElementClass);
+      const $el = getHTML('<div></div>');
+      $el.innerHTML = `<${tagName}></${tagName}>`;
+
+      $el.firstElementChild.setAttribute('foo', 'bar');
+
+      expect($el.firstElementChild.getAttribute('foo')).to.equal('bar');
+    });
+    it('should call removeAttribute', () => {
+      const {tagName, CustomElementClass} = getObservedAttributesTestElement(['foo']);
+      customElements.define(tagName, CustomElementClass);
+      const $el = getHTML('<div></div>');
+      $el.innerHTML = `<${tagName} foo></${tagName}>`;
+
+      $el.firstElementChild.removeAttribute('foo');
+
+      expect($el.firstElementChild.hasAttribute('foo')).to.be.false;
+    });
+    it('should call toggleAttribute', () => {
+      const {tagName, CustomElementClass} = getObservedAttributesTestElement(['foo']);
+      customElements.define(tagName, CustomElementClass);
+      const $el = getHTML('<div></div>');
+      $el.innerHTML = `<${tagName}></${tagName}>`;
+
+      $el.firstElementChild.toggleAttribute('foo', false);
+
+      expect($el.firstElementChild.hasAttribute('foo')).to.be.false;
+
+      $el.firstElementChild.toggleAttribute('foo', true);
```

Comment:
I think it would be good to explicitly set the attribute to the empty string before this `.toggleAttribute('foo', true)` to confirm that this call leaves the attribute there if it was already set.